### PR TITLE
#302 Camera resolution issue fixed

### LINKED ISSUE #302: [Bug] [CameraView] [Android] CameraView only creates low res (640x480) jpegs
### Description
The CameraView on Android only creates jpegs with a low resolution (640x480) while on iOS photos are captured with high resolution.
The images are also landscape while the screen is in portrait mode.

### Steps to Reproduce

1. Create a CameraView with CameraCaptureOptions.Photo
2. Trigger the shutter.

### Expected Behavior
High resolution photos are taken, depending on the device.

### Actual Behavior
Images are always in landscape mode and with a low resolution (640x480).

### PR DESCRIPTION
**Pull request by logeshpalani98**
### Bugs Fixed ###

- Fixes #302
Change Photo and resolution size dynamically. removed static photo size

` ImageReader.NewInstance(width: photoSize.Width, height: photoSize.Height, ImageFormatType.Jpeg, maxImages: 1);
`
### API Changes ###

<!-- List all API changes here (or just put None), example:

Added: 
 
- `string Class.Property { get; set; }`
- `void Class.Method();`

Changed:

 - `object Cell.OldPropertyName` => `object Cell.NewPropertyName`
 
If there is an entirely new API, then you can use a more verbose style:

```csharp
public static class NewClass {
    public static int SomeProperty { get; set; }
    public static void SomeMethod(string value);
}
``` -->

### Behavioral Changes ###

<!-- Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase. -->

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->

- [x] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs [hunk 1]
```diff
@@ -330,15 +333,15 @@ void SetupImageReader()
 		{
 			DisposeImageReader();
 
-			photoReader = ImageReader.NewInstance(640, 480, ImageFormatType.Jpeg, maxImages: 1);
+			photoReader = ImageReader.NewInstance(width: photoSize.Width, height: photoSize.Height, ImageFormatType.Jpeg, maxImages: 1);
```
Comment (VladislavAntonyuk) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:336]: is there a reason for adding property name for width and height but not for ImageFormatType. I think we can remove them at all

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs [hunk 2]
```diff
@@ -64,6 +65,8 @@ class CameraFragment : Fragment, TextureView.ISurfaceTextureListener
 		CameraTemplate cameraTemplate;
 		HandlerThread backgroundThread;
 		Handler backgroundHandler = null;
+
```
Comment (VladislavAntonyuk) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:68]: please remove extra spaces

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs [hunk 3]
```diff
@@ -221,7 +224,7 @@ public async Task RetrieveCameraDevice(bool force = false)
 					Element.MaxZoom = maxDigitalZoom = (float)characteristics.Get(CameraCharacteristics.ScalerAvailableMaxDigitalZoom);
 					activeRect = (Rect)characteristics.Get(CameraCharacteristics.SensorInfoActiveArraySize);
 					photoSize = GetMaxSize(map.GetOutputSizes((int)ImageFormatType.Jpeg));
-					videoSize = GetMaxSize(map.GetOutputSizes(Class.FromType(typeof(MediaRecorder))));
+					 videoSize = GetMaxSize(map.GetOutputSizes(Class.FromType(typeof(MediaRecorder))));
```
Comment (VladislavAntonyuk) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:227]: please remove extra space

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs [hunk 4]
```diff
@@ -428,7 +431,7 @@ public void StartRecord()
 			catch (Java.Lang.Exception error)
 			{
 				LogError("Failed to take video", error);
-				Element?.RaiseMediaCaptureFailed($"Failed to take video: {error}");
+				Element?.RaiseMediaCaptureFailed(message:$"Failed to take video: {error}");
```
Comment (VladislavAntonyuk) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:434]: please remove parameter name

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 1 above)
Comment (roubachof) [reply XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:336]: I tend to think that named parameters are great for non self explainable arguments.
In this case I will keep only the `maxImages` named parameter.

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 3 above)
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:227]:
```suggestion
					videoSize = GetMaxSize(map.GetOutputSizes(Class.FromType(typeof(MediaRecorder))));
```

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 4 above)
Comment (logeshpalani98) [reply XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:434]: removed

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 3 above)
Comment (logeshpalani98) [reply XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs]: removed

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 2 above)
Comment (logeshpalani98) [reply XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:68]: removed

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs (hunk 1 above)
Comment (logeshpalani98) [reply XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:336]: width and height param name removed

File: XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs [hunk 5]
```diff
@@ -569,7 +570,7 @@ public void UpdateRepeatingRequest()
 				if (cameraTemplate == CameraTemplate.Record)
 					sessionBuilder.Set(CaptureRequest.FlashMode, (int)flashMode);
 
-				session.SetRepeatingRequest(sessionBuilder.Build(), null, backgroundHandler);
+				session.SetRepeatingRequest(sessionBuilder.Build(), listener:null, backgroundHandler);
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs:573]:
```suggestion
				session.SetRepeatingRequest(sessionBuilder.Build(), listener: null, backgroundHandler);
```

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by jfversluis** [comment]
Great @logeshpalani98 thank you so much for fixing this bug! And welcome to the Xamarin Community Toolkit Contributors Club!

**Comment by logeshpalani98** [comment]
Thanks @jfversluis

**Pull request merged**